clients/venice-producer/src/main/java/com/linkedin/venice/producer/VeniceProducerMetrics.java

```diff
@@ -0,0 +1,85 @@
+package com.linkedin.venice.producer;
+
+import com.linkedin.venice.stats.AbstractVeniceStats;
+import io.tehuti.metrics.MetricsRepository;
+import io.tehuti.metrics.Sensor;
+import io.tehuti.metrics.stats.Avg;
+import io.tehuti.metrics.stats.Max;
+import io.tehuti.metrics.stats.OccurrenceRate;
+import io.tehuti.metrics.stats.Total;
+
+
+public class VeniceProducerMetrics extends AbstractVeniceStats {
+  private final boolean enableMetrics;
+  private Sensor requestSensor = null;
+  private Sensor putRequestSensor = null;
+  private Sensor deleteRequestSensor = null;
+  private Sensor updateRequestSensor = null;
+  private Sensor successRequestSensor = null;
+  private Sensor failedRequestSensor = null;
+  private Sensor produceLatencySensor = null;
+  private Sensor pendingRequestSensor = null;
+
+  public VeniceProducerMetrics(MetricsRepository metricsRepository, String storeName) {
+    super(metricsRepository, storeName);
+
+    if (metricsRepository != null) {
+      enableMetrics = true;
+      requestSensor = registerSensor("write_request", new OccurrenceRate());
+      putRequestSensor = registerSensor("put_request", new OccurrenceRate());
+      deleteRequestSensor = registerSensor("delete_request", new OccurrenceRate());
+      updateRequestSensor = registerSensor("update_request", new OccurrenceRate());
+
+      successRequestSensor = registerSensor("success_write_request", new OccurrenceRate());
+      failedRequestSensor = registerSensor("failed_write_request", new OccurrenceRate());
+      produceLatencySensor = registerSensor("produce_latency", new Avg(), new Max());
+
+      pendingRequestSensor = registerSensor("pending_write_request", new Total());
```

Comment:
Hmm... I wonder if it's ideal to have a `Total` here... I see that you're incrementing and decrementing it in the functions below, so it'll be accurate at any point in time, but as far as the metric is concerned, it will take just a single snapshot per minute, which could miss a lot of the action in-between data points. An alternative might be to define this metric as, let's say `Avg` and `Max`, and then outside of the metric, we'd keep an `AtomicInteger` and then call `pendingRequestSensor.record(pendingRequestAtomicCounter.incrementAndGet());` so that the metric gives us a sense of the range of values.

Reply:
Makes sense. Given a uniform load, it is better to get a statistic on the whole window rather than a point-in-time value